app/scripts/directives/apropiaciones/productos_apropiacion.js

```diff
@@ -87,17 +90,18 @@ angular.module('contractualClienteApp')
             }
 
             if (productosData.length > 0) {
-              productosData.forEach(function(uniqueProducto) {
-                if (uniqueProducto.Id !== productoSchema.Id) {
-                  productosData.push(productoSchema);
-                }
-              });
+              if(productosData.some(function(uniqueProducto) {
+                return productoSchema._id === uniqueProducto._id;
+              })){
+
+              } else {
+                productosData.push(productoSchema);
```

Comment:
Negando la condición del if se puede evitar dejar una de las ramas del if, vacías...
```suggestion
              if(!productosData.some(function(uniqueProducto) {
                return productoSchema._id === uniqueProducto._id;
              })){
                productosData.push(productoSchema);
```